subsys/mgmt/mcumgr/grp/settings_mgmt/src/settings_mgmt.c

```diff
@@ -0,0 +1,528 @@
+	}
+#endif
+
+	memcpy(key_name, key.value, key.len);
+	key_name[key.len] = 0;
+
+	if (IS_ENABLED(CONFIG_MCUMGR_GRP_SETTINGS_ACCESS_HOOK)) {
+		/* Send request to application to check if access should be allowed or not */
+		struct settings_mgmt_access settings_access_data = {
+			.access = SETTINGS_ACCESS_DELETE,
+			.name = key_name,
+		};
+
+		enum mgmt_cb_return status;
+		int32_t ret_rc;
+		uint16_t ret_group;
+
+		status = mgmt_callback_notify(MGMT_EVT_OP_SETTINGS_MGMT_ACCESS,
+					      &settings_access_data, sizeof(settings_access_data),
+					      &ret_rc, &ret_group);
+
+		if (status != MGMT_CB_OK) {
+			bool ok;
+
+			if (status == MGMT_CB_ERROR_RC) {
+				return ret_rc;
+			}
+
+			ok = smp_add_cmd_err(zse, ret_group, (uint16_t)ret_rc);
+			goto end;
+		}
+	}
+
+	/* Delete requested key from settings */
+	rc = settings_delete(key_name);
+
+#ifdef CONFIG_MCUMGR_GRP_SETTINGS_BUFFER_TYPE_HEAP
+	free(key_name);
+#endif
+
+	if (rc < 0) {
+		if (rc == -ENOENT) {
+			rc = SETTINGS_MGMT_ERR_KEY_NOT_FOUND;
+		} else {
+			rc = SETTINGS_MGMT_ERR_UNKNOWN;
+		}
+
+		ok = smp_add_cmd_err(zse, MGMT_GROUP_ID_SETTINGS, (uint16_t)rc);
+	}
+
+end:
+	return MGMT_RETURN_CHECK(ok);
+}
+
+/**
+ * Command handler: settings commit
+ */
+static int settings_mgmt_commit(struct smp_streamer *ctxt)
+{
+	bool ok = true;
+
+	if (IS_ENABLED(CONFIG_MCUMGR_GRP_SETTINGS_ACCESS_HOOK)) {
+		/* Send request to application to check if access should be allowed or not */
+		struct settings_mgmt_access settings_access_data = {
+			.access = SETTINGS_ACCESS_COMMIT,
+		};
+
+		zcbor_state_t *zse = ctxt->writer->zs;
+		enum mgmt_cb_return status;
+		int32_t ret_rc;
+		uint16_t ret_group;
+
+		status = mgmt_callback_notify(MGMT_EVT_OP_SETTINGS_MGMT_ACCESS,
+					      &settings_access_data, sizeof(settings_access_data),
+					      &ret_rc, &ret_group);
+
+		if (status != MGMT_CB_OK) {
+			bool ok;
+
+			if (status == MGMT_CB_ERROR_RC) {
+				return ret_rc;
+			}
+
+			ok = smp_add_cmd_err(zse, ret_group, (uint16_t)ret_rc);
+			goto end;
+		}
+	}
+
+	settings_commit();
+
+end:
+	return MGMT_RETURN_CHECK(ok);
+}
+
+/**
+ * Command handler: settings load
+ */
+static int settings_mgmt_load(struct smp_streamer *ctxt)
+{
+	bool ok = true;
+
+	if (IS_ENABLED(CONFIG_MCUMGR_GRP_SETTINGS_ACCESS_HOOK)) {
+		/* Send request to application to check if access should be allowed or not */
+		struct settings_mgmt_access settings_access_data = {
+			.access = SETTINGS_ACCESS_LOAD,
+		};
+
+		zcbor_state_t *zse = ctxt->writer->zs;
+		enum mgmt_cb_return status;
+		int32_t ret_rc;
+		uint16_t ret_group;
+
+		status = mgmt_callback_notify(MGMT_EVT_OP_SETTINGS_MGMT_ACCESS,
+					      &settings_access_data, sizeof(settings_access_data),
+					      &ret_rc, &ret_group);
+
+		if (status != MGMT_CB_OK) {
+			bool ok;
+
+			if (status == MGMT_CB_ERROR_RC) {
+				return ret_rc;
+			}
+
+			ok = smp_add_cmd_err(zse, ret_group, (uint16_t)ret_rc);
+			goto end;
+		}
+	}
+
+	settings_load();
```

Comment:
Is it ok to run `setting_load` from the subsystem? Bluetooth for example leave the responsability to call this to the application. Since this causes the callbacks to be run it could have some odd side effects I think  -- or at least I imagine that that's why bluetooth does not call it by itself.

Reply:
This is a command people can use remotely, so will only run if someone sends the settings load command from a MCUmgr client

Reply:
Oh I see that now, sorry I thought it was issued for every read. Cool makes sense then.